mlrun/launcher/local.py

```diff
@@ -45,171 +48,91 @@ def save(runtime):
+        project: Optional[str] = "",
+        params: Optional[dict] = None,
+        inputs: Optional[Dict[str, str]] = None,
+        out_path: Optional[str] = "",
+        workdir: Optional[str] = "",
+        artifact_path: Optional[str] = "",
+        watch: Optional[bool] = True,
         # TODO: don't use schedule from API schemas but rather from mlrun client
-        schedule: typing.Optional[
-            typing.Union[str, mlrun.api.schemas.schedule.ScheduleCronTrigger]
+        schedule: Optional[
+            Union[str, mlrun.api.schemas.schedule.ScheduleCronTrigger]
         ] = None,
         hyperparams: Dict[str, list] = None,
-        hyper_param_options: typing.Optional[mlrun.model.HyperParamOptions] = None,
-        verbose: typing.Optional[bool] = None,
-        scrape_metrics: typing.Optional[bool] = None,
-        local: typing.Optional[bool] = False,
-        local_code_path: typing.Optional[str] = None,
-        auto_build: typing.Optional[bool] = None,
-        param_file_secrets: typing.Optional[Dict[str, str]] = None,
-        notifications: typing.Optional[List[mlrun.model.Notification]] = None,
+        hyper_param_options: Optional[mlrun.model.HyperParamOptions] = None,
+        verbose: Optional[bool] = None,
+        scrape_metrics: Optional[bool] = None,
+        local: Optional[bool] = False,
+        local_code_path: Optional[str] = None,
+        auto_build: Optional[bool] = None,
+        param_file_secrets: Optional[Dict[str, str]] = None,
+        notifications: Optional[List[mlrun.model.Notification]] = None,
         returns: Optional[List[Union[str, Dict[str, str]]]] = None,
-    ):
-        mlrun.utils.helpers.verify_dict_items_type("Inputs", inputs, [str], [str])
-
-        if runtime.spec.mode and runtime.spec.mode not in run_modes:
-            raise mlrun.errors.MLRunInvalidArgumentError(
-                f'run mode can only be {",".join(run_modes)}'
-            )
+    ) -> "mlrun.run.RunObject":
 
         # do not allow local function to be scheduled
-        if schedule is not None:
+        if self._is_run_local and schedule is not None:
             raise mlrun.errors.MLRunInvalidArgumentError(
                 "local and schedule cannot be used together"
             )
 
         self._enrich_runtime(runtime)
-
         run = self._create_run_object(task)
 
-        local_function = self._create_local_function_for_execution(
+        if self._is_run_local:
+            runtime = self._create_local_function_for_execution(
+                runtime=runtime,
+                run=run,
+                local_code_path=local_code_path,
+                project=project,
+                name=name,
+                workdir=workdir,
+                handler=handler,
+            )
+
+        # sanity check
+        elif runtime._is_remote:
+            raise mlrun.errors.MLRunInvalidArgumentError(
+                "Remote function cannot be executed locally"
+            )
```

Comment:
not quite sure about that error message, as a "remote" runtime can be executed locally if passed with `local=True` I think you need to be more specific what exactly you are trying to catch, like what is being done with the schedule

Reply:
if `local=True` then `_is_run_local` is True as well and we will enter the 1st if clause and not this one so we are covered in that case.